src/channels.ml

```diff
@@ -77,13 +77,47 @@ let of_seekable_fd fd =
     return () in
   return { c with skip }
 
-let sslctx =
-  Ssl.init ();
-  Ssl.create_context Ssl.SSLv23 Ssl.Client_context
+let _ = Ssl.init ()
 
-let of_ssl_fd fd =
+let default_good_ciphers = "!EXPORT:RSA+AES128-SHA256"
+let default_legacy_ciphers = "RSA+AES256-SHA:RSA+AES128-SHA:RSA+RC4-SHA:RSA+DES-CBC3-SHA"
+
+let sslctx_cached = ref None
+
+let gen_sslctx legacy ciphers =
+  let ctx =
+    (* Here SSLv23 means ALL protocols *)
+    Ssl.create_context
+      (if legacy then Ssl.SSLv23 else Ssl.TLSv1_2)
```

Comment:
Why, you ask? I don't know why, but this is how the Ssl module works: see
https://github.com/savonet/ocaml-ssl/blob/master/src/ssl.mli#L205
and https://github.com/savonet/ocaml-ssl/blob/master/src/ssl.mli#L285
SSLv23 means "all protocols, both SSL and TLS" as an argument to the `create_context` function, but means only SSL protocols when passed as an argument to the `disable_protocols` function.

I tried to make that clear through my comments saying what my code achieves, but perhaps I should have been even more explicit.